var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { default as fetch } from 'node-fetch';
import { deposition_show_details } from '../deposition/show/details';
import { helpers_get_api } from '../helpers/get-api';
export const deposition_delete = (token, sandbox, version_id, verbose = false) => __awaiter(void 0, void 0, void 0, function* () {
    const msg = `deleting draft deposition with id ${version_id}...`;
    if (verbose) {
        console.log(msg);
    }
    yield deposition_show_details(token, sandbox, version_id, verbose);
    const api = helpers_get_api(sandbox);
    const endpoint = `/deposit/depositions/${version_id}`;
    const method = 'DELETE';
    const headers = {
        'Authorization': `Bearer ${token}`
    };
    const init = { method, headers };
    let response;
    response = yield fetch(`${api}${endpoint}`, init);
    if (response.ok !== true) {
        throw new Error(`(errid 9) Something went wrong on ${method} to ${api}${endpoint}: ${response.status} - ${response.statusText}`);
    }
    if (verbose) {
        console.log(`${msg}done`);
    }
});
